sympy/series/tests/test_gruntz.py

```diff
@@ -474,6 +475,17 @@ def test_issue_7096():
     from sympy.functions import sign
     assert gruntz(x**-pi, x, 0, dir='-') == oo*sign((-1)**(-pi))
 
+
+def test_issue_7391_8166():
+    f = Function('f')
+    y = Symbol('y')
+    func = x*y*y / (x*x + y**4)
+    func2 = func.subs(y, f(x))
```

Comment:
Rather than using subs here it would be clearer to write the actual expression:
```python
expr = x*f(x)**2 / (x**2 + f(x)**4)
```
In fact we can just put that expression directly in the test if we are not going to do anything else with it:
```python
raises(ValueError, lambda: gruntz(x*f(x)**2/(x**2 + f(x)**4), x, 0))
```